spec/ics-25-handler-interface/README.md

```diff
@@ -0,0 +1,365 @@
+```
+
+`confirmConnection` finalizes a connection (part three of the three-way connection handshake).
+
+The default IBC relayer module will allow external calls to `confirmConnection`.
+
+```coffeescript
+function confirmConnection(string identifier, CommitmentProof proofAck)
+```
+
+`timeoutConnection` handles a timeout in a connection opening handshake.
+
+```coffeescript
+function timeoutConnection(string identifier, CommitmentProof proofTimeout)
+```
+
+`queryConnection` queries an existing connection by known identifier, returning the associated metadata if found.
+
+```coffeescript
+function queryConnection(string identifier) -> Maybe<ConnectionInfo>
+```
+
+`initCloseConnection` initiates the graceful connection closing process. It will fail if there are any open channels using the connection or if the identifier is invalid.
+
+```coffeescript
+function initCloseConnection(string identifier) -> Maybe<Err>
+```
+
+`tryCloseConnection` continues the graceful connection closing process. It will fail if there are any open channels using the connection, if the proof is invalid, or if the identifier is invalid.
+
+The default IBC relayer module will allow external calls to `tryCloseConnection`.
+
+```coffeescript
+function tryCloseConnection(string identifier, proofInit) -> Maybe<Err>
+```
+
+`ackCloseConnection` finalizes the graceful connection closing process. It will fail if the proof is invalid or if the identifier is invalid.
+
+The default IBC relayer module will allow external calls to `ackCloseConnection`.
+
+```coffeescript
+function ackCloseConnection(string identifier, proofTry) -> Maybe<Err>
+```
+
+Implementations of `initConnection`, `tryConnection`, `ackConnection`, `confirmConnection`, `queryConnection`, `initCloseConnection`, `tryCloseConnection`, and `ackCloseConnection` are defined in ICS 3.
+
+### Channels
+
+By default, channels are owned by the creating module, meaning only the creating module can inspect, close, or send on the channel. A module can create any number of channels.
+
+`ChannelOptions` contains all the parameter choices required to create a new channel.
+
+```golang
+type ChannelOptions struct {
+  string          connectionIdentifier
+  bool            ordered
+}
+```
+
+`ChannelInfo` contains metadata about an existing channel.
+
+```golang
+type ChannelInfo struct {
+  string         channelIdentifier
+  ChannelOptions options
+}
+```
+
+`initChannel` tries to start the handshake to create a new channel with the provided options, failing if the connection is not found or the options are invalid.
+
+```coffeescript
+function initChannel(ChannelOptions options) -> Maybe<err>
+```
+
+`tryChannel` tries to initialize a channel based on proof of an initialization attempt on the counterparty chain, failing if the channel identifier is unavailable, the proof is invalid, or the calling module is not authorized.
+
+```coffeescript
+function tryChannel() -> Maybe<Err>
```

Comment:
Doesn't try channel (and others below) need a parameter (ChannelOptions or ChannelInfo) identifying the channel the request covers?

Reply:
Fixed, thanks.